### PR TITLE
Spec archive optimisation

### PR DESCRIPTION
**Pull request by vishal-biyani**
Spec archive has issues with single files (Will file issue in a bit) and can be optimized to use the Glob method already available.

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/1068)

### CODE REVIEW

**Comment by vishal-biyani** [comment]
Fixes https://github.com/fission/fission/issues/1070

File: fission/spec.go [hunk 1]
```diff
@@ -960,31 +960,11 @@ func applyResources(fclient *client.Client, specDir string, fr *FissionResources
...
-
-	// get a list of files from the include/exclude globs.
-	//
-	// XXX if there are lots of globs it's probably more efficient
-	// to do a filepath.Walk and call path.Match on each path...
 	files := make([]string, 0)
-	if len(aus.IncludeGlobs) == 1 && archiver.Zip.Match(aus.IncludeGlobs[0]) {
-		files = append(files, aus.IncludeGlobs[0])
-	}
-	for _, relativeGlob := range aus.IncludeGlobs {
```
Comment (life1347) [inline fission/spec.go]: The root cause of the issue is here. We can solve the issue by adding an `else` here.

File: fission/spec.go [hunk 2]
```diff
@@ -960,31 +960,11 @@ func applyResources(fclient *client.Client, specDir string, fr *FissionResources
 // localArchiveFromSpec creates an archive on the local filesystem from the given spec,
 // and returns its path and checksum.
 func localArchiveFromSpec(specDir string, aus *ArchiveUploadSpec) (*fission.Archive, error) {
-	// get root dir
-	var rootDir string
-	if len(aus.RootDir) == 0 {
```
Comment (life1347) [inline fission/spec.go]: No, we cannot do this. If we didn't add root dir to the file path as an absolute path, then we are not able to create the archive if the spec dir is not in the current path.
```
$ fission spec apply --specdir $GOPATH/src/github.com/fission/fission-bank-sample/specs
Fatal error: Failed to apply specs: Archive 'bank-go-zip' is empty
```

File: fission/spec.go (hunk 1 above)
Comment (vishal-biyani) [reply fission/spec.go]: Thanks - good catch - I have fixed and updated the PR @life1347

**Comment by life1347** [comment]
LGTM

**Pull request merged**